redfish-core/lib/log_services.hpp

```diff
@@ -3415,6 +3440,609 @@ inline void getSystemHardwareIsolationLogService(
+ *        and fill into "Message" property of LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ *
+ * @return The redfish response with "Message" property of LogEntry schema
+ *         if success else nothing in redfish response.
+ */
+
+inline void getPrettyNameByDbusObjPath(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const sdbusplus::message::object_path& dbusObjPath,
+    const size_t entryJsonIdx)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, dbusObjPath,
+         entryJsonIdx](const boost::system::error_code ec,
+                       const GetObjectType& objType) mutable {
+            if (ec || objType.empty())
+            {
+                BMCWEB_LOG_ERROR << "DBUS response error [" << ec.value()
+                                 << " : " << ec.message()
+                                 << "] when tried to get the dbus name"
+                                    "of isolated hareware: "
+                                 << dbusObjPath.str;
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if (objType.size() > 1)
+            {
+                BMCWEB_LOG_ERROR << "More than one dbus service implemented "
+                                    "the xyz.openbmc_project.Inventory.Item "
+                                    "interface to get the PrettyName";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if (objType[0].first.empty())
+            {
+                BMCWEB_LOG_ERROR << "The retrieved dbus name is empty for the "
+                                    "given dbus object: "
+                                 << dbusObjPath.str;
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            crow::connections::systemBus->async_method_call(
+                [asyncResp, dbusObjPath, entryJsonIdx](
+                    boost::system::error_code ec,
+                    const std::variant<std::string>& prettyName) mutable {
+                    if (ec)
+                    {
+                        BMCWEB_LOG_ERROR << "DBUS response error ["
+                                         << ec.value() << " : " << ec.message()
+                                         << "] when tried to get the "
+                                            "PrettyName of isolated hareware: "
+                                         << dbusObjPath.str;
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    if (auto prettyNameVal =
```

Comment:
need to check nullptr here

Reply:
Not required, since the implicit conversion will happen that will return true for non-null pointer and false for a null pointer.